ev3dev/core.py

```diff
@@ -182,8 +183,16 @@ def get_index(file):
                 self._device_index = get_index(name)
                 self.connected = True
             except StopIteration:
+                self._path = None
+                self._device_index = None
                 self.connected = False
 
+    def __str__(self):
```

Comment:
Does anyone else have comments on this? If we make this change it should be to cover a defect.